runtime/src/telemetry/mod.rs

```diff
@@ -1,3 +1,56 @@
 //! Utilities for collecting and reporting telemetry data.
 
 pub mod metrics;
+pub mod traces;
+
+use crate::tokio::Context;
+use crate::{Metrics, Spawner};
+use std::net::SocketAddr;
+use traces::exporter;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::Registry;
+
+/// Initialize telemetry with the given configuration.
+pub fn init(
```

Comment:
Although this is only implemented for tokio, I think it is still worth keeping in this folder.